src/main/java/org/junit/runner/OrderWith.java

```diff
@@ -0,0 +1,26 @@
+package org.junit.runner;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Inherited;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.junit.runner.manipulation.Ordering;
+
+/**
+ * When a test class is annotated with <code>&#064;OrderWith</code> or extends a class annotated
+ * with <code>&#064;OrderWith</code>, JUnit will order the tests in the test class (and child
+ * test classes, if any) using the ordering defined by the {@link Ordering} class.
+ *
+ * @since 4.13
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Inherited
+public @interface OrderWith {
+    /**
+     * @return a class that extends {@link Ordering} (must have a public no-argument constructor)
```

Comment:
I'm not sure why we would want end users to have to understand that filtering and sorting are applied via this new rule mechanism. They are orthogonal, and I think we should provide the simplest possible API to enable them.

I don't see why one couldn't implement `@FilterWith` today, without any changes outside of `ParentRunner`. Anyone extending `ParentRunner` would get that "for free", and filtering would work for children in a suite as long as the children implement `Filterable`. It sounds like what you are proposing would require a new general purpose mix-in interface, and users would have the confusing problem of having a `Runner` that's `Filterable` but can't be filtered via a `RunRule`

What you are proposing sounds like a larger redesign of runners, and one we should consider as part of the runner refactoring being proposed by `JUnitLambda`.